src/libs/actions/Report.ts

```diff
@@ -4329,6 +4332,31 @@ function completeOnboarding({
         });
     }
 
+    if (
+        !shouldSkipTestDriveModal &&
+        // Only add the dismissed state of the test drive modal when the user is not redirected to oldDot,
+        // because we don't want the modal to reappear when returning from oldDot.
+        !(engagementChoice === CONST.ONBOARDING_CHOICES.MANAGE_TEAM && willRedirectToOldDotFromOnboarding)
+    ) {
```

Comment:
I find the styling here a bit odd
```suggestion
    // Only add the dismissed state of the test drive modal when the user is not redirected to oldDot,
    // because we don't want the modal to reappear when returning from oldDot.
    if (!shouldSkipTestDriveModal && !(engagementChoice === CONST.ONBOARDING_CHOICES.MANAGE_TEAM && willRedirectToOldDotFromOnboarding)
    ) {
```